unstract/connectors/src/unstract/connectors/databases/mysql_handler.py

```diff
@@ -13,6 +13,16 @@
 
 logger = logging.getLogger(__name__)
 
+# MySQL/MariaDB connection error codes mapping
+CONNECTION_ERROR_MESSAGES = {
+    1045: "Authentication failed. Please check your username, password and ssl-settings enabled.",
+    2003: "Cannot connect to server. Please check the host, port, and network connectivity.",
+    1049: "Unknown database. Please verify the database name exists.",
+    2005: "Unknown host. Please check the host address.",
+    2006: "MySQL server has gone away. Connection was lost.",
```

Comment:
NIT: Gone away feels a bit unconventional, try this instead?
```suggestion
    2006: "MySQL server is unavailable. Connection was lost.",
```